test/replicate.js

```diff
@@ -781,3 +781,57 @@ test('non-sparse snapshot during partial replication', async function (t) {
   await s.update()
   t.is(s.contiguousLength, s.length)
 })
+
+test('sparse replication without gossiping', async function (t) {
+  const a = await create()
+  const b = await create(a.key)
+  const c = await create(a.key)
+
+  await a.append(['a', 'b', 'c'])
+
+  let s
+
+  s = replicate(a, b)
+  await b.download({ start: 0, end: 3 }).downloaded()
+  await unreplicate(s)
+
+  s = replicate(b, c)
+  await c.download({ start: 0, end: 3 }).downloaded()
+  await unreplicate(s)
+
+  await a.append(['d', 'e', 'f', 'd'])
+
+  s = replicate(a, b)
+  await b.download({ start: 4, end: 7 }).downloaded()
+  await unreplicate(s)
+
+  await t.test('block', async function (t) {
+    s = replicate(b, c)
+    t.teardown(() => unreplicate(s))
+
+    t.alike(await c.get(4), Buffer.from('e'))
+  })
+
+  await t.test('range', async function (t) {
+    s = replicate(b, c)
+    t.teardown(() => unreplicate(s))
+
+    await c.download({ start: 4, end: 6 }).downloaded()
+    t.pass('resolved')
+  })
+
+  await t.test('discrete range', async function (t) {
+    s = replicate(b, c)
+    t.teardown(() => unreplicate(s))
+
+    await c.download({ blocks: [4, 6] }).downloaded()
+    t.pass('resolved')
+  })
+
+  await t.test('seek', async function (t) {
```

Comment:
i think this fails if it happens as the first test here cuase the previous impact it

Reply:
Oh shoot, I should have created separate copies of the third core 🤦